include/LightGBM/utils/LocaleContext.h

```diff
@@ -0,0 +1,38 @@
+#ifndef __LOCALE_CONTEXT_H__
+#define __LOCALE_CONTEXT_H__
+
+#include <clocale>
+#include <locale>
+
+/**
+ * Class to override the program global locale during this object lifetime.
+ * After the object is destroyed, the locale is returned to its original state.
+ *
+ * @warn This is not thread-safe.
+*/
+class LocaleContext
+{
+public:
+
+  /**
+   * Override the current program global locale during this object lifetime.
+   *
+   * @param target_locale override the locale to this locale setting.
+   * @warn This is not thread-safe.
+   * @note This doesn't override cout, cerr, etc.
+   */
+  LocaleContext(const char* target_locale = "C") {
+    std::locale::global(std::locale(target_locale));
+  }
+
+  /**
+   * Restores the old global locale.
+   */
+  ~LocaleContext() {
+    std::locale::global(_saved_global_locale);
+  }
+private:
+  std::locale _saved_global_locale; //!< Stores global locale at initialization.
```

Comment:
```
At least two spaces is best between code and comments  [whitespace/comments] [2]
```
```suggestion
  std::locale _saved_global_locale;  // Stores global locale at initialization.
```

Reply:
This was to enrich the doxygen-style comments for the member data.

I'm applying the suggestion anyway.